flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -129,6 +134,33 @@ public static void setupYarnClassPath(Configuration conf, Map<String, String> ap
 		}
 	}
 
+	/**
+	 * Setup and register a pre-uploaded resource according to the file status.
+	 *
+	 * @param localSrcPath
+	 * 		path to the local file
+	 * 		relative target path of the file (will be prefixed be the full home directory we set up)
+	 * @param preUploadedFileStatus
+	 *    the corresponding pre-uploaded file status of localSrcPath
+	 *
+	 * @return Path to remote file (usually hdfs)
+	 */
+	static Tuple2<Path, LocalResource> setupPreUploadedResource(
```

Comment:
@TisonKun I am not sure if there's a follow up JIRA created yet for PR #10187 ? are you planning to use similar approach like this one to support a remote URL resource (like Jars)? if yes I think this is fine, otherwise `setUpYarnPublicCacheResource` might be more accurate?

Reply:
The preUploaded Resources means they have been uploaded to hdfs already, not Yarn public cache.
I have to make clear that Yarn public cache is a type of `LocalResourceVisibility`. The public cache will be downloaded only once in the same NodeManager and could be shared by different applications.

Reply:
Not yet. I don't have too much cycles for generally thinking a considerate solution of this demand now. Will revisit it later.